app/controllers/authentication_controller.rb

```diff
@@ -0,0 +1,79 @@
+class AuthenticationController < ApplicationController
+  before_action :check_if_enabled
+
+  MISSING_EMAIL_ERROR = 'Please enter your email address.'.freeze
+  INVALID_EMAIL_ERROR = 'This doesn’t look like a valid email address – check you’ve entered it correctly.'.freeze
+
+  def sign_in
+    @address = params[:address]
+  end
+
+  def request_sign_in_token
+    unless params[:address].present?
+      flash.now[:error] = MISSING_EMAIL_ERROR
+      return render :sign_in
+    end
+
+    @address = params.require(:address)
+
+    email_alert_api.create_auth_token(
+      address: @address,
+      destination: process_sign_in_token_path
+    )
+  rescue GdsApi::HTTPUnprocessableEntity
+    flash.now[:error] = INVALID_EMAIL_ERROR
+    return render :sign_in
+  end
+
+  def process_sign_in_token
+    subscriber_id, redirect = read_token(params.require(:token))
+
+    if subscriber_id.nil?
+      @sign_in_link = sign_in_path
+      return render :expired_link
+    end
+
+    authenticate_subscriber(subscriber_id)
+
+    destination = safe_redirect_destination(redirect) || '/email/manage'
```

Comment:
We should change this to use a rails route helper method I think it's `list_subscriptions_path`

Reply:
Done